llvm/lib/Target/AArch64/AArch64RegisterInfo.cpp

```diff
@@ -552,7 +552,9 @@ bool AArch64RegisterInfo::hasBasePointer(const MachineFunction &MF) const {
     if (hasStackRealignment(MF))
       return true;
 
-    if (MF.getSubtarget<AArch64Subtarget>().hasSVE()) {
+    auto &ST = MF.getSubtarget<AArch64Subtarget>();
+    if (ST.hasSVE() ||
+        (ST.hasSME() && ST.hasStreamingInterfaceOrBody(MF.getFunction()))) {
```

Comment:
Oops, didn't realize there was an existing bit for this... `ST,isStreaming()` is equivalent, I think.

Reply:
Don't we need to check it on a per Function basis?

Reply:
The subtarget is per-function, sort of... see AArch64TargetMachine::getSubtargetImpl.

Reply:
Oh right, of course.